benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -872,18 +875,21 @@ void GraphicsBenchmarkApp::ProcessKnobs()
 {
     // Detect if any knob value has been changed
     // Note: DigestUpdate should be called only once per frame (DigestUpdate unflags the internal knob variable)!
-    const bool allTexturesTo1x1KnobChanged = pAllTexturesTo1x1->DigestUpdate();
-    const bool alphaBlendKnobChanged       = pAlphaBlend->DigestUpdate();
-    const bool depthTestWriteKnobChanged   = pDepthTestWrite->DigestUpdate();
-    const bool enableSpheresKnobChanged    = pEnableSpheres->DigestUpdate();
+    const bool allTexturesTo1x1KnobChanged    = pAllTexturesTo1x1->DigestUpdate();
+    const bool alphaBlendKnobChanged          = pAlphaBlend->DigestUpdate();
+    const bool depthTestWriteKnobChanged      = pDepthTestWrite->DigestUpdate();
+    const bool enableSpheresKnobChanged       = pEnableSpheres->DigestUpdate();
+    const bool sphereInstanceCountKnobChanged = pSphereInstanceCount->DigestUpdate();
 
     const bool enableSpheres = pEnableSpheres->GetValue();
     // If the LOD is empty, assuming we skipped the setup for all sphere resources at start
     // So need to do the intial setup for all resources here
-    const bool spheresAreSetUp               = mSpheresAreSetUp;
-    const bool updateSphereDescriptors       = spheresAreSetUp && allTexturesTo1x1KnobChanged;
-    const bool setupSphereResourcesAndMeshes = (!spheresAreSetUp) && enableSpheres;
-    const bool rebuildSpherePipeline         = setupSphereResourcesAndMeshes || (spheresAreSetUp && (alphaBlendKnobChanged || depthTestWriteKnobChanged));
+    const bool spheresAreSetUp         = mSpheresAreSetUp;
+    const bool requireMoreSpheres      = sphereInstanceCountKnobChanged && (pSphereInstanceCount->GetValue() > mInitializedSpheres);
+    const bool updateSphereDescriptors = spheresAreSetUp && allTexturesTo1x1KnobChanged;
+    const bool setupSphereResources    = (!spheresAreSetUp) && enableSpheres;
+    const bool setupSphereMeshes       = ((!spheresAreSetUp) || requireMoreSpheres) && enableSpheres;
+    const bool rebuildSpherePipeline   = setupSphereMeshes || (spheresAreSetUp && (alphaBlendKnobChanged || depthTestWriteKnobChanged));
```

Comment:
This is a bit hard to follow, and used later in the code where you have to scroll up to figure out when something is rebuilt. Also it's a little weird to update `mSpheresAreSetUp` in `SetupSphereMeshes()`, since mSpheresAreSetUp basically means everything is setup instead of just spheres. So if I call just `SetupSpheresMeshes()` mSpheresAreSetUp will be true. 
Maybe not for this PR, but we should  probably consider cleaning up the logic a bit, one alternative: https://github.com/google/bigwheels/commit/9e5c0b8319a8f6503d3631cee46285aab3c68fa5

Reply:
Thank you for the comment. I applied your suggested clean up and the code looks more readable now.